peerprovider/http.go

```diff
@@ -0,0 +1,26 @@
+package peerprovider
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"golang.org/x/net/context"
+)
+
+type httpPeerProvider struct{}
+
+func (httpPeerProvider) Resolve(ctx context.Context, url *url.URL) ([]string, error) {
+	resp, err := http.Get(url.String())
+	if err != nil {
+		return nil, fmt.Errorf("failed to read peer list over HTTP: %v", err)
+	}
```

Comment:
we should close the response body once we're done with it `defer resp.Body.Close()`